pipeline/transformer/transformer_tei/src/main/scala/weco/pipeline/transformer/tei/transformers/TeiSubjects.scala

```diff
@@ -0,0 +1,65 @@
+package weco.pipeline.transformer.tei.transformers
+
+import weco.catalogue.internal_model.identifiers.IdState.{
+  Identifiable,
+  Unidentifiable
+}
+import weco.catalogue.internal_model.identifiers.{
+  IdState,
+  IdentifierType,
+  SourceIdentifier
+}
+import weco.catalogue.internal_model.work.{Concept, Subject}
+import weco.pipeline.transformer.tei.NormaliseText
+
+import scala.xml.{Elem, Node}
+
+object TeiSubjects {
+
+  /**
+    * Subjects live in the profileDesc block of the tei which looks like this:
+    * <profileDesc>
+    *   <textClass>
+    *     <keywords scheme="#LCSH">
+    *       <list>
+    *         <item>
+    *           <term ref="subject_sh85083116">Medicine, Arab</term>
+    *         </item>
+    *       </list>
+    *     </keywords>
+    *   </textClass>
+    * </profileDesc>
+    *
+    */
+  def apply(xml: Elem): List[Subject[IdState.Unminted]] =
+    (xml \\ "profileDesc" \\ "keywords").flatMap { keywords =>
+      val identifierType = (keywords \@ "scheme").toLowerCase.trim match {
+        case s if s == "#lcsh" => Some(IdentifierType.LCSubjects)
+        case s if s == "#mesh" => Some(IdentifierType.MESH)
+        case _                 => None
+      }
+      (keywords \\ "term").flatMap { term =>
+        val label = NormaliseText(term.text)
+        val reference = parseReference(term)
+        val id = (reference, identifierType) match {
+          case (Some(r), Some(identifierType)) =>
+            Identifiable(SourceIdentifier(identifierType, "Subject", r))
+          case _ => Unidentifiable
+        }
+        label.map(l =>
+          Subject(id = id, label = l, concepts = List(Concept(label = l))))
+      }
+    }.toList
+
+  private def parseReference(term: Node) = {
+    val referenceString = term \@ "ref"
+    // arabic manuscripts seem to have the subject id in the
+    // attribute "key" instead of "ref" ¯\_(ツ)_/¯
+    val idValue =
+      if (referenceString.isEmpty) term \@ "key" else referenceString
+    // some of the subject ids are prepended with "subject_sh" for lcsh or " subject" for mesh.
+    // So here we remove that to get the id
```

Comment:
Is the 'sh' part of the LCSH identifier, or do we want to bin that too? It's not obvious to me.

e.g. in your example is the identifier `sh85083116` or `85083116`

Reply:
Yeah that comment was written poorly, I've updated it.
Yes, for lcsh `sh` is part of the id so we keep it , we only discard the `subject_` bit before